### PR TITLE
Penny-Pincher: A budget app

### PR DESCRIPTION
**Pull request by Abe1able**
### In this project I:

- Created a Ruby on Rails application that allows the user to:
    > register and log in, so that the data is private to them.
    > introduce new transactions associated with a category.
    > see the money spent on each category.

- Added authentication with devise
- Added unit and integration test for main components
- Used controllers to handle requests and render empty views.
- Used params from browser requests in a safe way.
- Used preprocessed html file with embedded Ruby code.
- Used layouts and templates for shared content.
- Used database migration files to maintain database schema.
- Used validations for models.
- Secured app from n+1 problems.

Here is the live demo link. [Video](https://www.loom.com/share/7edd18c43a814a98bd436c02acbb33f8?sid=d33b755a-3c2b-436b-ac68-0bb836c6ab52)

And here is the deployed app. [Live](https://abe1able-penny-pincher.onrender.com)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,175 @@
...
+![Preview](./app/assets/images/erd_diagram.png)
+
+
+### Live Demo
+This is a [live](https://abe1able-penny-pincher.onrender.com) demo of the mobile app
+
+
+### Video Demo
+This is a [video](https://www.loom.com/share/05c9a959d9054528a726f479e885aaa1?sid=2f006c21-4acd-4adb-9891-bbbc34f71bb1) demonstration of the app
+
```
Comment (Alfredbis29) [inline README.md]: - [ ]You recorded an amazing video so far but we still need to improve on it according to the requirements :wink: ,  Please revise the video to include the following changes: provide a demo of the project showcasing all its features and include a highlight of a code snippet you find interesting or are proud of, while ensuring the video remains within a maximum length of 5 minutes and maintains a clear and objective description in professional English.

**Review by Alfredbis29 — CHANGES_REQUESTED** [review]
#  STATUS: REQUIRED CHANGES 
 ## Hi @Abe1able 
 Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

  
 ## Highlights 
:heavy_check_mark:  Your Work looks Professional 
:heavy_check_mark:  Nice work so far :+1: 
 
 ## Required Changes 

- [x]  Check Under the review

 ## Optional suggestions 

- [ ] N/A

 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
 
 Cheers and Happy coding! :clap: :clap: :clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (Abe1able) [reply README.md]: Thank you so much for the review. I have recorded a new [video](https://www.loom.com/share/7edd18c43a814a98bd436c02acbb33f8?sid=d33b755a-3c2b-436b-ac68-0bb836c6ab52) including the changes you suggested.

**Review by Alfredbis29 — APPROVED** [review]
# STATUS: APPROVED   
 ## Hi  @Abe1able 
Your project is complete! There is nothing else to say other than... Great job clap Congratulations! :tada:
 
 TO HIGHLIGHT:
 
 * Your project is professional :white_check_mark:
 * Well Setup Your Test :white_check_mark:
 * Your Project is well structured :white_check_mark: 
  * Good Job :+1: :smile: 
 
 Now you can merge this branch and move on 1st_place_medal: HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :


 
 ## Optional suggestions

- [x] Keep doing the Great work :clinking_glasses: 

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
 
Cheers and Happy coding! :clap:  :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Abe1able** [comment]
Thank you so much @Alfredbis29

**Pull request merged**